### PR TITLE
[12.0][IMP] add job helper for simplifying testing

### PR DESCRIPTION
**Pull request by rousseldenis**
Cherry-picked 6f007dbb19485e8d06f14dd55f534095379af22d

### CODE REVIEW

**Comment by rousseldenis** [comment]
@sebastienbeau @lmignon

**Comment by rousseldenis** [comment]
Any news of this ?

**Review by guewen — APPROVED** [review]

File: queue_job/tests/common.py [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+        return len(self.existing)
+
+    def search_created(self):
+        return self.search_all() - self.existing
+
+    def search_all(self):
+        return self.env['queue.job'].search([])
+
+
+class JobMixin(unittest.TestCase):
```
Comment (simahawk) [inline queue_job/tests/common.py]: the base class here probably is irrelevant

**Review by simahawk — APPROVED** [review]
We should probably document this together w/ https://github.com/OCA/queue/issues/152 and the usage of `test_queue_job_no_delay` ctx key.

File: queue_job/tests/common.py (hunk 1 above)
Comment (rousseldenis) [reply queue_job/tests/common.py]: I admit I've just cherry-picked (and it works) :smiley:

Comment (simahawk) [reply queue_job/tests/common.py]: sure it works :) I'm just saying that you have to use this mixin together w/ an odoo test class which already inherits from base unittest class.
Plus, here you need and env to work, which means you cannot really use this class standalone. 
Anyway, is just a detail...

**Comment by lmignon** [comment]
@rousseldenis What's the status of this one? Do you plan to fix the last comment?

**Comment by simahawk** [comment]
@rousseldenis ping :)

File: queue_job/tests/common.py (hunk 1 above)
Comment (sbidoul) [inline queue_job/tests/common.py]:
```suggestion
class JobMixin:
```

File: queue_job/tests/common.py [hunk 2]
```diff
@@ -0,0 +1,35 @@
+import unittest
+from odoo.addons.queue_job.job import Job
+
+
+class JobCounter(object):
```
Comment (sbidoul) [inline queue_job/tests/common.py]:
```suggestion
class JobCounter:
```

File: queue_job/tests/common.py (hunk 1 above)
Comment (sbidoul) [reply queue_job/tests/common.py]: So I made 2 suggestions (no need to inherit from `object` in python 3)

**Comment by rousseldenis** [comment]
@simahawk @sbidoul Done

**Review by sebastienbeau — APPROVED** [review]
LGTM, thanks for porting it

**Pull request merged**